src/github/client/github-client-server.test.ts

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable jest/no-standalone-expect */
+import { getLogger } from "config/logger";
+import { GitHubInstallationClient } from "./github-installation-client";
+import { statsd }  from "config/statsd";
+import { InstallationId } from "./installation-id";
+import nock from "nock";
+import { AppTokenHolder } from "./app-token-holder";
+import fs from "fs";
+import { envVars }  from "config/env";
+import { GitHubEnterpriseUrls } from "~/src/util/check-github-app-type";
+jest.mock("config/feature-flags");
+
+describe("GitHub Client", () => {
+	const githubInstallationId = 17979017;
+	let statsdHistogramSpy;
+
+	beforeEach(() => {
+		// Lock Time
+		statsdHistogramSpy = jest.spyOn(statsd, "histogram");
+	});
+
+	afterEach(() => {
+		// Unlock Time
+		statsdHistogramSpy.mockRestore();
+	});
+
+	function givenGitHubReturnsPullrequests(
+		owner: string,
+		repo: string,
+		perPage: number,
+		page: number,
+		expectedInstallationTokenInHeader?: string,
+		scope: nock.Scope = githubNock
+	) {
+		scope
+			.get(`/repos/${owner}/${repo}/pulls`)
+			.query({
+				per_page: perPage,
+				page,
+				installationId: /^.*$/
+			})
+			.matchHeader(
+				"Authorization",
+				expectedInstallationTokenInHeader
+					? `Bearer ${expectedInstallationTokenInHeader}`
+					: /^Bearer .+$/
+			)
+			.matchHeader("Accept", "application/vnd.github.machine-man-preview+json")
+			.reply(200, [
+				{ number: 1 } // we don't really care about the shape of this response because it's in GitHub's hands anyways
+			]);
+	}
+
+	function verifyMetricsSent(path: string, status) {
+		expect(statsdHistogramSpy).toBeCalledWith("app.server.http.request.github", expect.anything(), expect.objectContaining({
+			client: "axios",
+			method: "GET",
+			path,
+			status
+		}));
+	}
+
+	it.skip("works with a non-cloud installation", async () => {
```

Comment:
Was going to add more tests but this PR is already getting pretty big so, seeing everything is hidden behind a feature flag, decided I'm going to create a separate ticket to address tests for GHE server.